Feature/post about TDD with vue (#9)
altixco/altixco.github.io: _posts/2020-11-09-example-tdd-vue.md

---
layout: post
title: Basic TDD with Vue
tags: vuejs, tdd
author: Sebastian Peralta
---

> TDD is a practice where you write test before code, this with the end of have the best product possible, the advantage has it is forces you to think before you code. the cycle of TDD is the next:

Reviewer: ```suggestion
TDD is a practice where you write tests before coding, this purpose of TDD is to have the best possible product, its main advantage is that it forces you to think before you code. the cycle of TDD is the next:
```

>   1. you describe a test which an expected behaivor. (it is will fail)

Reviewer: ```suggestion
  1. you describe a test with the expected behavior. (here it should fail, the failing is important to ensure both states of the test, the *failure* state, and the *success* state)
```

>   2. you write code simple that solution the test.

Reviewer: ```suggestion
  2. you code a simple solution that solves the test.
```

>   3. refactor code for to make rigth.

Reviewer: ```suggestion
  3. refactor the code to make it professional 👌 
```

> quick example (vue, jest):

Reviewer: Please use Markdown subtitles in these cases.

```suggestion
### Quick example (vue, jest):
```

> - We will create a component in vue that displays a list of colors. The first thing to do would be to write the proof corresponding to this logic:

Reviewer: ```suggestion
- We will create a component in vue that displays a list of colors. The first thing to do would be to write the test corresponding to this logic:
```
